### PR TITLE
Loop against this.items.length not this.items

### PR DESCRIPTION
**Pull request by gkatsev**

### CODE REVIEW

**Comment by dmlap** [comment]
I wish we had a test that would have failed for this. Otherwise, LGTM

**Comment by gkatsev** [comment]
I'll try and come up with one.

File: test/videojs-playlist-ui.test.js [hunk 1]
```diff
@@ -313,6 +313,38 @@ test('updates on "playlistchange"', function() {
   equal(items.length, playlist.length, 'updated with the new items');
 });
 
+test('updates on "playliastchange", equal lengths', function() {
```
Comment (gkatsev) [inline test/videojs-playlist-ui.test.js]: this test will fail for the forloop that was fixed.

**Comment by gkatsev** [comment]
@dmlap updated. Now we have a test for each of the 3 conditions in `update()`.

**Comment by gkatsev** [comment]
For some reason playlist returns `-1` for `currentItem` in the failing test.

**Comment by dmlap** [comment]
LGTM

**Pull request merged**